apis/v1alpha1/gatewayclass_types.go

```diff
@@ -77,7 +81,7 @@ type GatewayClassSpec struct {
 	// Support: Custom
 	//
 	// +optional
-	ParametersRef *LocalObjectReference `json:"parametersRef,omitempty"`
+	ParametersRef LocalObjectReference `json:"parametersRef,omitempty"`
```

Comment:
optional should be pointer?

Reply:
Per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required:
> They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or _have a built-in nil value (e.g. maps and slices)._

Take a look through other areas of Service APIs and k8s native APIs and you will see:
```
// +optional
SomeField SomeFieldCustomType `json:"someField,omitempty"`
```

Reply:
> They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) __or have a built-in nil value (e.g. maps and slices).__

Reply:
I don't think that `ParametersRef` has an OK nil value; the empty string fields create ambiguity that could easily be avoided.

Reply:
I will revert these changes so the PR can proceed.